actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -145,19 +248,24 @@ def render_to_body(options)
     end
 
     def _render_to_body_with_renderer(options)
-      _renderers.each do |name|
-        if options.key?(name)
-          _process_options(options)
-          method_name = Renderers._render_with_renderer_method_name(name)
-          return send(method_name, options.delete(name), options)
-        end
+      _renderers.each do |renderer_name|
+        next unless options.key?(renderer_name)
+        _process_options(options)
+
+        renderer_target_method_name = Renderers._render_with_renderer_method_name(renderer_name)
+        renderer_target_value = options.delete(renderer_name)
+        serializer_name = options.key?(:serializer_name) ? options.delete(:serializer_name).to_sym : renderer_name
+        serialized_value = _serializers[serializer_name].call(renderer_target_value, options)
```

Comment:
` _render_to_body_with_renderer` 

1. If there is a renderer for the format
1. Get the renderer method name and object to render
1. :new: get the serializer_name from options else same name as renderer/format
1. :new: serialize object using serializer with that name
1. send serialized object to renderer method (:new: which no longer modifies the object)